src/python/pants/backend/python/util_rules/interpreter_constraints.py

```diff
@@ -66,8 +67,19 @@ def parse_constraint(constraint: str) -> Requirement:
     """
     try:
         parsed_requirement = Requirement.parse(constraint)
-    except ValueError:
-        parsed_requirement = Requirement.parse(f"CPython{constraint}")
+    except ValueError as err:
+        if "CPython" in constraint:
+            raise InvalidRequirement(
+                f"Failed to parse Python interpreter constraint `{constraint}`: {err.args[0]}"
+            )
+
+        try:
+            parsed_requirement = Requirement.parse(f"CPython{constraint}")
+        except ValueError as err:
+            raise InvalidRequirement(
+                f"Failed to parse Python interpreter constraint `{constraint}`: {err.args[0]}"
+            )
```

Comment:
`InvalidRequirement` is a subclass of `ValueError` so one cannot handle `ValueError` before `InvalidRequirement` so we check whether it has failed to parse because it's a shorthand (by trying to parse with `CPython` prefix added).

Reply:
What do you think about removing the `if "CPython" in constraint:` codepath, and throw this one referring to the _outer_ error, with what the user originally wrote?

e.g.

```python
try:
    ...
except ValueError as err:
    try:
        ...
    except ValueError: # CHANGE: no `err` shadowing here
        raise InvalidRequirement(...)
```

Reply:
Great suggestion, thanks for this, this is now done!